usr/share/rear/output/USB/Linux-i386/100_create_efiboot.sh

```diff
@@ -51,8 +51,34 @@ mkdir -p $efi_dst || Error "Failed to create directory '$efi_dst'"
 # "cp: failed to preserve ownership for '/tmp/rear-efi.XXXXXXXXXX/EFI/BOOT/kernel': Operation not permitted"
 # because it copies to a VFAT filesystem on the EFI partition (see format/USB/default/300_format_usb_disk.sh)
 # cf. https://github.com/rear/rear/issues/2683
-# Copy boot loader:
-cp -L $v "$UEFI_BOOTLOADER" "$efi_dst/BOOTX64.efi" || Error "Failed to copy UEFI_BOOTLOADER '$UEFI_BOOTLOADER' to $efi_dst/BOOTX64.efi"
+# Copy UEFI bootloader:
+if test -f "$SECURE_BOOT_BOOTLOADER" ; then
+    # For a technical description of Shim see https://mjg59.dreamwidth.org/19448.html
+    # Shim is a signed EFI binary that is a first stage bootloader
+    # that loads and executes another (signed) EFI binary
+    # which normally is a second stage bootloader
+    # which normally is a GRUB EFI binary
+    # which normally is available as a file named grub*.efi
+    # so when SECURE_BOOT_BOOTLOADER is used as UEFI_BOOTLOADER
+    # (cf. rescue/default/850_save_sysfs_uefi_vars.sh)
+    # then Shim (usually shim.efi) must be copied as EFI/BOOT/BOOTX64.efi
+    # and Shim's second stage bootloader must be also copied where Shim already is.
+    DebugPrint "Using '$SECURE_BOOT_BOOTLOADER' as first stage Secure Boot bootloader BOOTX64.efi"
+    cp -L $v "$SECURE_BOOT_BOOTLOADER" "$efi_dst/BOOTX64.efi" || Error "Failed to copy SECURE_BOOT_BOOTLOADER '$SECURE_BOOT_BOOTLOADER' to $efi_dst/BOOTX64.efi"
+    # When Shim is used, its second stage bootloader can be actually anything
```

Comment:
Is this code copied from output/ISO/Linux-i386/250_populate_efibootimg.sh ? It looks a bit different (e.g. the `-L` argument to `cp` and different variable names).

Reply:
I mean:
When we have Secure Boot working
for ISO within the existing ISO code
and for USB within the existing USB code,
then - as a separated second step -
the whole mess could be cleaned up (if possible).

Reply:
Could you please add a comment explaining whre is the code copied from? Also, would it be feasible to reduce differences between the two versions? For example, why `cp -L` in one version and mere `cp` in other? Or, `test -f "$SECURE_BOOT_BOOTLOADER" || build_bootx86_efi $efi_dst/BOOTX64.efi $efi_dst/grub.cfg "/boot" "$UEFI_BOOTLOADER"` on one version and
```sh
if ! test -f "$SECURE_BOOT_BOOTLOADER" ; then
    build_bootx86_efi $TMP_DIR/mnt/EFI/BOOT/BOOTX64.efi $efi_boot_tmp_dir/grub.cfg "$boot_dir" "$UEFI_BOOTLOADER"
fi
```
in the other.

I am asking for this because if this code needs a cleanup (it does), it will be easier to do if both places to be cleaned up look as similar as possible.

Reply:
@pcahyna 
basically you ask me for some cleanup right now.
I will have a look if I could do something right now.

`cp -L` was already there in the existing USB code
so I used it there also for my new code.

`cp -L` was not used in the existing ISO code
so I did not introduce it there.

On first glance `cp -L` looks better than plain `cp`
but I don't know the reason behind why it is different
and right now I do not like to investigate about this.
This belongs to a later cleanup of that whole mess.

Reply:
@pcahyna 
with
https://github.com/rear/rear/commit/efb37fb9b535c9d04d97d9711cc05945972ced8f
I did what you asked for except the `cp -L` cleanup
(e.g. using `cp -L` also in 250_populate_efibootimg.sh).

Reply:
@jsmeix thanks, it looks much clearer now

Reply:
@pcahyna 
my pleasure!